### PR TITLE
Add list sites assigned to hostname API

### PR DESCRIPTION
**Pull request by naveedaz**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/CommonDefinitions.json [hunk 1]
```diff
@@ -956,6 +956,47 @@
...
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Identifier"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string"
+        }
```
Comment (dsgouda) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/CommonDefinitions.json]: Please mark this as readonly

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/CommonDefinitions.json [hunk 2]
```diff
@@ -956,6 +956,47 @@
     }
    }
   },
+    "Identifier": {
```
Comment (dsgouda) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/CommonDefinitions.json]: nit: fix indentation

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 3]
```diff
@@ -254,6 +254,41 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Web/listSitesAssignedToHostName": {
+      "post": {
+        "summary": "List all apps that are assigned to a hostname.",
+        "description": "List all apps that are assigned to a hostname.",
+        "operationId": "ListSitesAssignedToHostName",
```
Comment (dsgouda) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: Incorrect operationId pattern, ideally this should be
`List_Identifiers` or maybe `List_IdentifiersAssignedToHostName`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Minor comments need to be addressed

**Comment by dsgouda** [comment]
Please move the API version in `stable` directory if it is not preview

**Comment by naveedaz** [comment]
rebasing properly

**Pull request closed without merging**